### ISSUE TITLE
When using ping_interval, if the connection drops and reconnects, the pinging stops

### ISSUE THREAD

**Issue by juliangall** [websocket-client/websocket-client#867]
When making a connection for the first time, specifying a ping, the ping thread is started before setSock() is called in run_forever():

```python
        if ping_interval:
            event = threading.Event()
            thread = threading.Thread(
                target=self._send_ping, args=(ping_interval, event, ping_payload))
            thread.daemon = True
            thread.start()

        setSock()
````

However, after a disconnection, setSock() is called again to reconnect, but the ping thread is no longer running as it was killed by the disconnection. A new thread needs to be started.

I have fixed this by removing the thread code above from before the initial setSock() call, and putting it instead at the start of the setSock() function, which now looks like this:

```python
        def setSock(reconnecting=False):
            nonlocal thread
            if not (thread and thread.is_alive()):
                if ping_interval:
                    event = threading.Event()
                    thread = threading.Thread(
                        target=self._send_ping, args=(ping_interval, event, ping_payload))
                    thread.daemon = True
                    thread.start()
            self.sock = WebSocket(
                self.get_mask_key, sockopt=sockopt, sslopt=sslopt,
                fire_cont_frame=self.on_cont_message is not None,
                skip_utf8_validation=skip_utf8_validation,
                enable_multithread=True)
        ...
```

The thread therefore gets started for the initial connection, and also for subsequent reconnections.

Julian

**Comment by j-a-n** [comment]
Could you please try if #862 fixes the problem?

**Comment by juliangall** [comment]
I have tried #862 and, yes, it fixes my problem. I haven't tested anything else it does though.

Julian

**Comment by engn33r** [comment]
#862 has been merged so v1.5.0 should not have this issue.
